ci instability: conbenchlegacy failing to collect tests

On `main`:

```
_________________ ERROR collecting tests/benchmark/test_cli.py _________________
legacy/tests/benchmark/test_cli.py:236: in <module>
    register_benchmarks(this_dir)
legacy/conbenchlegacy/util.py:154: in register_benchmarks
    import_path(f"{directory}/{filename}", root=entry)
E   TypeError: import_path() missing 1 required keyword-only argument: 'consider_namespace_packages'
=========================== short test summary info ============================
ERROR legacy/tests/benchmark/test_cli.py - TypeError: import_path() missing 1 required keyword-only argument: 'consider_namespace_packages'
```

https://github.com/conbench/conbench/actions/runs/8021686867/job/22838333924#step:10:21

--- a/legacy/requirements-cli.txt
+++ b/legacy/requirements-cli.txt
@@ -4,3 +4,4 @@ psutil
 py-cpuinfo
 PyYAML
 requests
+pytest<8.1

